library/ssl_msg.c

```diff
@@ -2102,7 +2102,7 @@ int mbedtls_ssl_flush_output( mbedtls_ssl_context *ssl )
         if( ret <= 0 )
             return( ret );
 
-        if( (size_t)ret > ssl->out_left || ( INT_MAX > SIZE_MAX && ret > SIZE_MAX ) )
+        if( (size_t)ret > ssl->out_left || ( INT_MAX > SIZE_MAX && (size_t)ret > SIZE_MAX ) )
```

Comment:
Apologies for not spotting this earlier, but this should be `ret > (int) SIZE_MAX`. The condition `(size_t) ret > SIZE_MAX` is vacuous.

Reply:
Fixed in 9230f97.